sweep/sweeper.go

```diff
@@ -649,6 +643,13 @@ func (s *UtxoSweeper) collector(blockEpochs <-chan *chainntnfs.BlockEpoch) {
 		case input := <-s.newInputs:
 			s.handleNewInput(input)
 
+			// If this input is forced, we perform an sweep
+			// immediately.
+			if input.params.Force {
```

Comment:
Hmm, what if we split the middle a bit here? At times we want to force sweep outgoing HTLCs that may be expiring soon. IIUC, we'll still progressively start to batch those as they come in, but if we have `N` outgoing HTLCs at force close time, then we end up publishing `N` distinct transactions here. Are we at risk of inadvertently violating RBF rules if we don't bump each increment by enough total fees?

Reply:
Yeah seems like "force" really means "set deadline to next block" and every time this loop fires we can recompute that. This would still allow for batching them.

Reply:
So the idea is we can eventually remove the usage of `Force` or even remove the param `Force`. We currently use it for three cases:
- when sweeping anchor with a deadline less than `anchorSweepConfTarget` which is 144.
- when sweeping the second-level timeout HTLCs.
- when bump force close fee via `bumpforceclosefee` RPC.

For the first two cases, and especially the second case, we do it to avoid cascading FCs, which served as a temp mitigation before #7683 is fixed. The last one is a bit different as it's user-triggered action, maybe they just want an FC to confirm even when it's uneconomical.

In other words, we previously used `force` to express multiple preferences - sweep immediately, sweep uneconomically, or bump fee and sweep. Now it just means sweep immediately and uneconomically. When we are able to fail HTLC backwards, we then can bundle the `N` HTLCs if they share the same `nLockTime`.